torch/lib/THD/master_worker/master/generic/THDTensorMath.cpp

```diff
@@ -21,6 +21,132 @@ void THDTensor_(zero)(THDTensor *r) {
   THDTensor_(fill)(r, 0);
 }
 
+void THDTensor_(maskedFill)(THDTensor *tensor, THDByteTensor *mask, real value) {
+  masterCommandChannel->sendMessage(
+    packMessage(
+      Functions::tensorMaskedFill,
+      tensor,
+      mask,
+      value
+    ),
+    THDState::s_current_worker
+  );
+}
+
+void THDTensor_(maskedCopy)(THDTensor *tensor, THDByteTensor *mask, THDTensor* src) {
+  if (THDTensor_(nElement)(tensor) != THDByteTensor_nElement(mask))
+    THError("Number of elements of destination tensor != Number of elements in mask");
+  masterCommandChannel->sendMessage(
+    packMessage(
+      Functions::tensorMaskedCopy,
+      tensor,
+      mask,
+      src
+    ),
+    THDState::s_current_worker
+  );
+}
+
+void THDTensor_(maskedSelect)(THDTensor *tensor, THDTensor* src, THDByteTensor *mask) {
+  ptrdiff_t numel = THDByteTensor_sumall(mask);
+  THDTensor_(resize1d)(tensor, numel);
+  masterCommandChannel->sendMessage(
+    packMessage(
+      Functions::tensorMaskedSelect,
+      tensor,
+      src,
+      mask
+    ),
+    THDState::s_current_worker
+  );
+}
+
+void THDTensor_(nonzero)(THDLongTensor *subscript, THDTensor *tensor) {
+  masterCommandChannel->sendMessage(
+    packMessage(
+      Functions::tensorNonzero,
+      subscript,
+      tensor
+    ),
+    THDState::s_current_worker
+  );
+  ptrdiff_t numel = receiveValueFromWorker<ptrdiff_t>(tensor->storage->node_id);
```

Comment:
Won't this break if `sizeof(ptrdiff_t) != sizeof(long long)` or they're represented differently? Use the same datatype on both ends...